packages/playwright-core/src/utils/browserFetcher.ts

```diff
@@ -63,3 +66,45 @@ export function logPolitely(toBeLogged: string) {
   if (!logLevelDisplay)
     console.log(toBeLogged);  // eslint-disable-line no-console
 }
+
+let cachedUserAgent: string | undefined;
+function getDownloadUserAgent(): string {
+  if (cachedUserAgent)
+    return cachedUserAgent;
+  try {
+    cachedUserAgent = determineDownloadUserAgent();
+  } catch (e) {
+    cachedUserAgent = 'Playwright/unknown';
+  }
+  return cachedUserAgent;
+}
+
+function determineDownloadUserAgent(): string {
+  let osIdentifier = 'unknown';
+  let osVersion = 'unknown';
+  if (process.platform === 'win32') {
+    const version = os.release().split('.');
+    osIdentifier = 'windows';
+    osVersion = `${version[0]}.${version[1]}`;
+  } else if (process.platform === 'darwin') {
+    const version = execSync('sw_vers -productVersion').toString().trim().split('.');
+    osIdentifier = 'macOS';
+    osVersion = `${version[0]}.${version[1]}`;
+  } else if (process.platform === 'linux') {
+    const version = getUbuntuVersionSync();
```

Comment:
I would generalize it to reading NAME + VERSION off /etc/os-release. That way we get all vendors. Not sure why getUbuntuVersionSync is doing more than that btw, /etc/os-release has all the info we need. While looking at it, I thought that maybe putting OS + architecture as a Playwright property is not what we want, maybe we want

Playwright/1.18 Ubuntu/20.04 (cpu=aarch)

Reply:
> I would generalize it to reading NAME + VERSION off /etc/os-release. That way we get all vendors. 

Turns out to be mostly true! https://gist.github.com/aslushnikov/8ceddb8288e4cf9db3039c02e0f4fb75